core/modules/widgets/text.js

```diff
@@ -31,7 +31,9 @@ TextNodeWidget.prototype.render = function(parent,nextSibling) {
 	this.computeAttributes();
 	this.execute();
 	var text = this.getAttribute("text",this.parseTreeNode.text || "");
+	var length = this.getAttribute("length");
```

Comment:
The default parameter is undefined. If you provide undefined to slice as the second parameter it will take the entire string, which is what we want in that case.